File: 2/2-1.py
import sys

if len(sys.argv) < 2:
    file = sys.stdin
else:
    file = open(sys.argv[1], 'rt')

safe = 0
for line in file:
    data = [int(x) for x in line.split()]
    sign = data[1] - data[0]
    if not sign:
        continue
    sign /= abs(sign)
    issafe = 1
    for index in range(len(data)-1):
        delta = (data[index+1] - data[index]) / sign
        if delta < 1 or delta > 3:
            issafe = 0
            break
    safe += issafe
    
print(safe)

File: 2/2-1.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <string>
#include <sstream>

using namespace std;


vector<int> split(string s)
{
    vector<int> data{};
    istringstream stream(s);
    int value;
    while (stream >> value)
        data.push_back(value);
    return data;
}

void main(int argc, char* argv[])
{
    istream &input = argc > 1 ? static_cast<istream&>(ifstream{argv[1]}) : cin;

    int safe{0};

	while(!input.eof()) {
        string s{};
	    getline(input, s);
	    if (!s.length())
	        break;
	    auto data = split(s);

        bool issafe{true};

	    if (data.size() >= 2) {
            auto sign = data[1] - data[0];
            if (!sign)
                continue;
            sign /= abs(sign);

            for (int i = 0; issafe && i < data.size() - 1; ++i) {
                auto delta = (data[i+1]-data[i]) / sign;
                issafe = (delta >= 1 && delta <= 3);
            }

        }
        if (issafe)
            ++safe;
	}

	cout << safe << endl;
}
